data/src/main/java/com/example/data/di/DataSourceModule.kt

```diff
@@ -6,20 +6,19 @@ import com.example.domain.datasource.PostsLocalDataSource
 import com.example.domain.datasource.PostsRemoteDataSource
 import dagger.Binds
 import dagger.Module
-import javax.inject.Singleton
+import dagger.Reusable
+import dagger.hilt.InstallIn
+import dagger.hilt.android.components.ApplicationComponent
 
-@Module(includes = [
-    ServiceModule::class,
-    PersistenceModule::class,
-    MapperModule::class
-])
+@Module
+@InstallIn(ApplicationComponent::class)
 internal abstract class DataSourceModule {
 
     @Binds
-    @Singleton
+    @Reusable
```

Comment:
Why would you make this reusable instead of SIngleton, this change is good but it is important to understand the difference